widget/scroller.go

```diff
@@ -306,52 +326,59 @@ func (r *scrollContainerRenderer) handleShadowVisibility(offset int, contentSize
 }
 
 func (r *scrollContainerRenderer) updatePosition() {
-	scrollWidth := r.scroll.Size().Width
-	contentWidth := r.scroll.Content.Size().Width
-	scrollHeight := r.scroll.Size().Height
-	contentHeight := r.scroll.Content.Size().Height
-	r.handleAreaVisibility(contentWidth, scrollWidth, r.horizArea)
-	r.handleAreaVisibility(contentHeight, scrollHeight, r.vertArea)
+	scrollSize := r.scroll.Size()
+	contentSize := r.scroll.Content.Size()
 
 	r.scroll.Content.Move(fyne.NewPos(-r.scroll.Offset.X, -r.scroll.Offset.Y))
-	r.handleShadowVisibility(r.scroll.Offset.X, contentWidth, scrollWidth, r.leftShadow, r.rightShadow)
-	r.handleShadowVisibility(r.scroll.Offset.Y, contentHeight, scrollHeight, r.topShadow, r.bottomShadow)
 
-	Renderer(r.vertArea).Layout(r.scroll.size)
-	Renderer(r.horizArea).Layout(r.scroll.size)
+	if r.scroll.Direction != scrollHorizontalOnly {
+		r.handleAreaVisibility(contentSize.Height, scrollSize.Height, r.vertArea)
+		r.handleShadowVisibility(r.scroll.Offset.Y, contentSize.Height, scrollSize.Height, r.topShadow, r.bottomShadow)
+		Renderer(r.vertArea).Layout(r.scroll.size)
+	}
+	if r.scroll.Direction != scrollVerticalOnly {
+		r.handleAreaVisibility(contentSize.Width, scrollSize.Width, r.horizArea)
+		r.handleShadowVisibility(r.scroll.Offset.X, contentSize.Width, scrollSize.Width, r.leftShadow, r.rightShadow)
+		Renderer(r.horizArea).Layout(r.scroll.size)
+	}
 
-	canvas.Refresh(r.vertArea)  // this is required to force the canvas to update, we have no "Redraw()"
-	canvas.Refresh(r.horizArea) // this is required like above but if we are horizontal
+	if r.scroll.Direction != scrollHorizontalOnly {
+		canvas.Refresh(r.vertArea) // this is required to force the canvas to update, we have no "Redraw()"
+	}
+	if r.scroll.Direction != scrollVerticalOnly {
+		canvas.Refresh(r.horizArea) // this is required like above but if we are horizontal
+	}
 }
 
 // ScrollContainer defines a container that is smaller than the Content.
 // The Offset is used to determine the position of the child widgets within the container.
 type ScrollContainer struct {
 	BaseWidget
-
-	Content fyne.CanvasObject
-	Offset  fyne.Position
+	Direction scrollDirection
```

Comment:
I think `scrollDirection` type should be public (and constants too), or `Direction` field should be private. Because in current implementation user can't assign anything to `Direction`. I like first option more.

Reply:
Thanks @pymq, these types and constants are now public.